notifier.cpp

```diff
@@ -83,11 +83,18 @@ int main(int argc, char** argv) {
     natsOptions_Destroy(opts);
     stanConnOptions_Destroy(connOpts);
 
-    while (!done && s == NATS_OK) {
+    while (s == NATS_OK) {
         auto msg = std::make_unique<message>();
         static const auto start = "{\"msg_type\":\"";   // ok, it's ugly. TODO: ?parse json
         static const auto start_len = strlen(start);
         std::getline(std::cin, msg->data);
+        if (done) {
+            if (msg->data.size()) {
+                std::cerr << "WARNING! Pipe hasn't empty." << std::endl;
+                done = false;
```

Comment:
This doesn't work because you clear `done` flag. User send `SIGTERM` signal, `done`-flag was set in interrupt handler. In this line it was clear. How way user can stop the notifier if `msg` has data?